src/hugr.rs

```diff
@@ -111,13 +111,15 @@ impl NodeType {
 
     /// Use the input extensions to calculate the concrete signature of the node
     pub fn signature(&self) -> Option<Signature> {
-        self.input_extensions
-            .as_ref()
-            .map(|rs| self.op.signature().with_input_extensions(rs.clone()))
+        self.input_extensions.as_ref().and_then(|rs| {
```

Comment:
I kept pondering whether there's a "right" order to check these in but failed to come up with one. How about
`if let (Some(rs), Some(sig)) = (self.input_extensions, self.op.signature()) {sig.with_input_extensions(rs.clone())} else {None}`

Reply:
I'm going to change this to keep this as previous behaviour (with an unwrap_or_default), I think changes to the inference logic should be a follow on